### DISCUSSION TITLE
Command Prompt vs Windows Terminal

### DISCUSSION THREAD

**Discussion by bcalco** [spectreconsole/spectre.console#686] [Q&A] [answered]
In Windows Terminal (wt.exe), my console app behaves as I expect. However, when run from the standard Command Prompt (such as the one created when VS runs it in the debugger), I get strange behavior. The app runs a DLR language via its concept of a ConsoleHost, and the REPL experience is greatly diminished by the behavior.

In a nutshell, when hitting ENTER three times, instead of

```
>
>
>
>
```
I get:
```
>
    >
        >
            >
```
It gets even worse when something besides ENTER is typed; it seems to indent from where the last character ended, plus two spaces.

I see there is a mode where I can turn on some kind of legacy behavior, but when I do, the app crashes in the Windows Terminal.

How do I make peace bweeen these worlds? I'm not sure I can mandate everyone install Windows Terminal (corporate environments often impose idiotic rules).

I'm exploring creating my own REPL experience from scratch using RadLine, on the theory that the ConsoleHost in the DLR is what's not playing nice. However, it would be good to know if there is a workaround till then.

**Comment by patriksvensson** [comment, marked as answer]
@bcalco We would need to see a minimal reproducible example to help you out here.
